internal/transport/http2_client.go

```diff
@@ -1628,7 +1636,13 @@ func (t *http2Client) readServerPreface() error {
 // network connection.  If the server preface is not read successfully, an
 // error is pushed to errCh; otherwise errCh is closed with no error.
 func (t *http2Client) reader(errCh chan<- error) {
-	defer close(t.readerDone)
+	var errClose error
+	defer func() {
+		close(t.readerDone)
+		if errClose != nil {
+			t.Close(errClose)
```

Comment:
Won't we end up calling t.Close() multiple times in this case? One another instance where we are making t.Close is [this](https://github.com/grpc/grpc-go/pull/7666/files#diff-5334018444bcb34d15bcb3c8a3050cdf9c8a40fe41305c3a602ea4bfeb0cd178R420). Thoughts?

Reply:
Yes I think `t.Close()` might be called multiple times but it will execute only once always becuase we check the state of the transport beofre actually executing anything, look here ,
 https://github.com/grpc/grpc-go/blob/bcf9171a20e44ed81a6eb152e3ca9e35b2c02c5d/internal/transport/http2_client.go#L991
 And also, `t.Close` was already being called inside the `reader` go routine , I just changed it to specifically be called only after the `t,readerDone` channel is closed, or it was going into infinite wait.